pkg/controller/che/che_controller.go

```diff
@@ -470,20 +477,19 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return nil, nil
 	}
 
+	pluginRegistryURL := instance.Spec.Server.PluginRegistryUrl
 	// Create Plugin registry resources unless an external registry is used
 	externalPluginRegistry := instance.Spec.Server.ExternalPluginRegistry
 	if !externalPluginRegistry {
-		pluginRegistryURL, err := addRegistryRoute("plugin")
+		guessedPluginRegistryURL, err := addRegistryRoute("plugin")
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 		if cheFlavor != "codeready" {
-			pluginRegistryURL += "/v3"
+			guessedPluginRegistryURL += "/v3"
```

Comment:
Should this be externalized as a property somehow? like a default registry version variable, rather than encoded this way?

Reply:
this could probably be externalized, but since it was already like that before this PR, I would let this for a next PR, no ?

Reply:
yeah, save it for later refactoring. I'll assume you'll open an issue for externalizing params and refactoring "keycloak" to "identityProvider" and target it for Che 7.2 or later.